book_haven_api/views.py

```diff
@@ -24,12 +26,38 @@ def get_a_library(request, library_id):
             "detail": "The library with id " + str(library_id) + " does not exist."
         }]},status=status.HTTP_404_NOT_FOUND)
     
-@api_view(['GET'])
+@api_view(['GET', 'POST'])
 def get_library_books(request, library_id):
-    library = Library.objects.get(id=library_id)
-    books = library.book_set
-    serializer = BookSerializer(books, many=True)
-    return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    if request.method == 'GET':
+        library = Library.objects.get(id=library_id)
+        books = library.book_set
+        serializer = BookSerializer(books, many=True)
+        return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    
+    elif request.method == 'POST':
+        library = Library.objects.get(id=library_id)
+        api_key = config('MY_API_KEY')
+        url = 'https://www.googleapis.com/books/v1/volumes?q=isbn:' + request.data['isbn'] + '&key=' + api_key
+        response = requests.get(url)
+        if response.json()['totalItems'] == 0:
+            return Response({"errors": [{
+                "status": "422",
+                "title": "Book not created",
+                "detail": "Book could not be created because ISBN not found."
+            }]},status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        
+        else:
+            book = Book.objects.create(
+                isbn=request.data['isbn'],
+                book_image=response.json()['items'][0]['volumeInfo']['imageLinks']['thumbnail'],
+                description=response.json()['items'][0]['volumeInfo']['description'],
+                title=response.json()['items'][0]['volumeInfo']['title'],
+                author=response.json()['items'][0]['volumeInfo']['authors'][0],
+                genre=response.json()['items'][0]['volumeInfo']['categories'][0],
+                library=library,
+            )
+            serializer = BookSerializer(book, many=False)
```

Comment:
One approach you can use is to pass the response to the serializer first, have it validate the info, and then use `.save` to actually create the record. That way you get the built in validation of the serializer before you even try to create anything.